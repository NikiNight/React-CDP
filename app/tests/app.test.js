import React from 'react';
import renderer from 'react-test-renderer';
import App from './../../app/components/app';
import {mount} from 'enzyme';

describe('Test App component', () => {

    test('matches snapshot', () => {
        const tree = renderer.create(<App/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should disable isMovieSelected state and set selectedMovieId to null', () => {
        const component = mount(<App/>);
        component.instance().backToSearch();
        expect(component.instance().state.isMovieSelected).toBe(false);
        expect(component.instance().state.selectedMovieId).toBe(null);
    });

    it('should enable isMovieSelected state and set selectedMovieId to 1', () => {
        const component = mount(<App/>);
        component.instance().selectMovie(1);
        expect(component.instance().state.isMovieSelected).toBe(true);
        expect(component.instance().state.selectedMovieId).toBe(1);
    });

})