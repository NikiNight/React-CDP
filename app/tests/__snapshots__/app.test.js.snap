// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component matches snapshot 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="wrapper"
    >
      <div
        className="header-line"
      >
        <a
          className="logo"
          href="#"
        >
          netflixroulette
        </a>
      </div>
      <div
        className="movie-search"
      >
        <h1>
          Find your movie
        </h1>
        <input
          className="movie-search__input"
          name="movie"
          type="text"
        />
        <div
          className="movie-search__buttons-line"
        >
          <div
            className="movie-search__buttons"
          >
            <span>
              Search by
            </span>
            <button
              className="button button_active"
              onClick={[Function]}
              value="Title"
            >
              Title
            </button>
            <button
              className="button "
              onClick={[Function]}
              value="Genre"
            >
              Genre
            </button>
          </div>
          <button
            className="button button_search"
            onClick={[Function]}
          >
            Search
          </button>
        </div>
      </div>
    </div>
  </header>,
  <main>
    <div
      className="sort-line"
    >
      <div
        className="wrapper"
      >
        <p
          className="sort-line__movies-num"
        >
          4
           movies found
        </p>
        <div
          className="sort-line__selectors"
        >
          Sort by
          <span
            className="button_minimal button_minimal_active"
            onClick={[Function]}
          >
            release date
          </span>
          <span
            className="button_minimal "
            onClick={[Function]}
          >
            raiting
          </span>
        </div>
      </div>
    </div>
    <div
      className="wrapper"
    >
      <div
        className="movies-list"
      >
        <article
          className="movie"
          onClick={[Function]}
        >
          <div
            className="movie__image-wrapper"
          >
            <img
              alt=""
              className="movie__image"
              src="https://image.tmdb.org/t/p/w500/aMETsaNNcDc6g5ZatQtVbySnSaA.jpg"
            />
          </div>
          <div
            className="movie__description"
          >
            <div>
              <h2
                className="movie__title"
              >
                Untitled Avengers
              </h2>
              <p
                className="movie__category"
              >
                Action, Adventure, Science Fiction
              </p>
            </div>
            <p
              className="movie__year"
            >
              2019
            </p>
          </div>
        </article>
        <article
          className="movie"
          onClick={[Function]}
        >
          <div
            className="movie__image-wrapper"
          >
            <img
              alt=""
              className="movie__image"
              src="https://image.tmdb.org/t/p/w500/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg"
            />
          </div>
          <div
            className="movie__description"
          >
            <div>
              <h2
                className="movie__title"
              >
                Avengers: Infinity War
              </h2>
              <p
                className="movie__category"
              >
                Adventure, Science Fiction, Fantasy, Action
              </p>
            </div>
            <p
              className="movie__year"
            >
              2018
            </p>
          </div>
        </article>
        <article
          className="movie"
          onClick={[Function]}
        >
          <div
            className="movie__image-wrapper"
          >
            <img
              alt=""
              className="movie__image"
              src="https://image.tmdb.org/t/p/w500/t90Y3G8UGQp0f0DrP60wRu9gfrH.jpg"
            />
          </div>
          <div
            className="movie__description"
          >
            <div>
              <h2
                className="movie__title"
              >
                Avengers: Age of Ultron
              </h2>
              <p
                className="movie__category"
              >
                Action, Adventure, Science Fiction
              </p>
            </div>
            <p
              className="movie__year"
            >
              2015
            </p>
          </div>
        </article>
        <article
          className="movie"
          onClick={[Function]}
        >
          <div
            className="movie__image-wrapper"
          >
            <img
              alt=""
              className="movie__image"
              src="https://image.tmdb.org/t/p/w500/cezWGskPY5x7GaglTTRN4Fugfb8.jpg"
            />
          </div>
          <div
            className="movie__description"
          >
            <div>
              <h2
                className="movie__title"
              >
                The Avengers
              </h2>
              <p
                className="movie__category"
              >
                Science Fiction, Action, Adventure
              </p>
            </div>
            <p
              className="movie__year"
            >
              2012
            </p>
          </div>
        </article>
      </div>
    </div>
  </main>,
  <footer
    className="footer"
  >
    <div
      className="wrapper"
    >
      <a
        className="logo"
        href="#"
      >
        netflixroulette
      </a>
    </div>
  </footer>,
]
`;
